Replace words that may disturb easily-offended users or easily-worried lawyers
with those that may entertain the easily-amused Sun engineer who had to edit
the list.

Do not remove unless you've gotten approval to ship with R-rated language
randomly displayed on screen.   (While you could argue that out of context
some of these words are, for instance, another name for a rooster or a cat,
it's harder to argue when the context they're displayed in may include words
like "boobies", "vibrator" and "Viagra".)

Unlikely to be accepted upstream.
---
 hacks/barcode.c | 127 +++++++++++++++++++++++++++++++++++++++++++++++++-------
 1 file changed, 113 insertions(+), 14 deletions(-)

diff --git a/hacks/barcode.c b/hacks/barcode.c
index d659216..b883c67 100644
--- a/hacks/barcode.c
+++ b/hacks/barcode.c
@@ -102,44 +102,66 @@ static const char *words[] =
   "addiction",
   "alertness",
   "Algeria",
+  "Amber Road",
   "anxiety",
   "aorta",
   "argyle socks",
   "attrition",
+  "automagic",
   "axis of evil",
   "bamboo",
+  "banana slug",
   "bangle",
   "bankruptcy",
   "baptism",
+  "barrelmaker",
   "beer",
   "bellicosity",
   "bells",
   "belly",
+  "Beowawe",
+  "Berkeley",
+  "BFU",
+  "Bleaklow",
   "bliss",
   "bogosity",
-  "boobies",
-  "boobs",
+  "bolthole",
+  "Boole House",
+  "boomer",
   "booty",
   "bread",
+  "brickify",
+  "browncoat",
   "bubba",
   "burrito",
+  "caiman",
   "California",
   "capybara",
   "cardinality",
   "caribou",
   "carnage",
+  "Chicago",
   "children",
+  "chime",
   "chocolate",
+  "chupacabra",
+  "cinnabar",
+  "Clear View",
   "CLONE",
-  "cock",
+  "Colorado",
+  "Complete.",
   "constriction",
   "contrition",
   "cop",
+  "corona",
   "corpse",
   "cowboy",
   "crabapple",
   "craziness",
+  "crossbow",
   "cthulhu",
+  "cuddle",
+  "cuddletech",
   "Death",
   "decepticon",
   "deception",
@@ -148,22 +170,31 @@ static const char *words[] =
   "decoy",
   "defenestration",
   "democracy",
+  "dendrite",
   "dependency",
   "despair",
   "desperation",
   "disease",
   "disease",
   "doberman",
+  "dot com",
   "DOOM",
-  "dreams",
+  "DREAM",
   "dreams",
   "drugs",
+  "Duckwater",
+  "duraflame",
   "easy",
   "ebony",
+  "ecoresponsibility",
   "election",
   "eloquence",
+  "Elko",
+  "Ely",
+  "embiggen",
   "emergency",
   "eureka",
+  "every city",
   "excommunication",
   "fat",
   "fatherland",
@@ -171,18 +202,29 @@ static const char *words[] =
   "fear",
   "fever",
   "filth",
+  "fishworks",
+  "flask",
   "flatulence",
   "fluff",
   "fnord",
+  "FOX",
+  "frak",
   "freedom",
   "fruit",
-  "fruit",
+  "fugu",
   "futility",
+  "galaxy",
+  "genunix",
   "gerbils",
-  "GOD",
+  "Gerlach",
   "goggles",
   "goobers",
   "gorilla",
+  "greenline",
+  "green violet",
+  "grok",
+  "grub",
+  "Groom Lake",
   "halibut",
   "handmaid",
   "happiness",
@@ -193,23 +235,31 @@ static const char *words[] =
   "heroin",
   "heroine",
   "hope",
+  "honeycomb",
   "hysteria",
+  "Ichthyosaur",
   "icepick",
   "identity",
   "ignorance",
+  "illumonati",
   "importance",
+  "Indiana",
   "individuality",
   "inkling",
   "insurrection",
   "intoxicant",
   "ire",
   "irritant",
+  "iwashi",
   "jade",
   "jaundice",
+  "Jet",
   "Joyce",
+  "Jupiter",
   "kidney stone",
   "kitchenette",
   "kiwi",
+  "Lahontan",
   "lathe",
   "lattice",
   "lawyer",
@@ -218,32 +268,49 @@ static const char *words[] =
   "lobbyist",
   "love",
   "lozenge",
+  "Lund",
+  "Mad Hatter",
   "magazine",
   "magnesium",
   "malfunction",
+  "Marrakesh",
+  "Mars",
   "marmot",
   "marshmallow",
+  "McGill",
+  "Menlo Park",
+  "mercurial",
   "merit",
   "merkin",
   "mescaline",
   "milk",
   "mischief",
   "mistrust",
+  "Moapa",
   "money",
   "monkey",
   "monkeybutter",
+  "Muskoka",
+  "mustang",
   "nationalism",
   "nature",
+  "Nevada",
+  "newt",
+  "Niagara",
   "neuron",
   "noise",
   "nomenclature",
   "nutria",
   "OBEY",
   "ocelot",
+  "oracle",
   "offspring",
   "overseer",
+  "Pahrump",
   "pain",
   "pajamas",
+  "panic",
+  "paravon",
   "passenger",
   "passion",
   "Passover",
@@ -253,19 +320,21 @@ static const char *words[] =
   "petticoat",
   "pharmacist",
   "PhD",
+  "pinenut",
+  "Pioche",
   "pitchfork",
   "plague",
   "Poindexter",
   "politician",
   "pony",
+  "ponytail",
   "presidency",
   "prison",
   "prophecy",
-  "Prozac",
   "punishment",
   "punk rock",
   "punk",
-  "pussy",
+  "quahog",
   "quagmire",
   "quarantine",
   "quartz",
@@ -274,13 +343,18 @@ static const char *words[] =
   "rage",
   "readout",
   "reality",
-  "rectum",
+  "register",
   "reject",
   "rejection",
+  "Reno",
   "respect",
   "revolution",
   "roadrunner",
+  "rock",
   "rule",
+  "sagebrush",
+  "Santa Clara",
+  "Santa Cruz",
   "savor",
   "scab",
   "scalar",
@@ -289,25 +363,36 @@ static const char *words[] =
   "security",
   "sediment",
   "self worth",
+  "shiny",
   "sickness",
+  "sierra",
   "silicone",
+  "Sirius",
   "slack",
   "slander",
   "slavery",
   "sledgehammer",
-  "smegma",
   "smelly socks",
+  "soda",
+  "solder",
+  "Sonoma",
+  "songbird",
   "sorrow",
   "space program",
+  "Sparks",
   "stamen",
   "standardization",
   "stench",
+  "stilt",
   "subculture",
   "subversion",
+  "sun",
   "suffering",
   "surrender",
   "surveillance",
   "synthesis",
+  "tamarack",
+  "Tahoe",
   "television",
   "tenant",
   "tendril",
@@ -315,17 +400,27 @@ static const char *words[] =
   "terrorism",
   "terrorist",
   "the impossible",
+  "the possimpible",
   "the unknown",
+  "thumper",
+  "tiger",
+  "tonic",
   "toast",
+  "Tonopah",
   "topography",
   "truism",
+  "truthiness",
+  "Tsinghua",
   "turgid",
+  "uisce",
+  "unbreakable",
   "underbrush",
   "underling",
   "unguent",
   "unusual",
   "uplink",
   "urge",
+  "Utah",
   "valor",
   "variance",
   "vaudeville",
@@ -333,8 +428,7 @@ static const char *words[] =
   "vegetarian",
   "venom",
   "verifiability",
-  "viagra",
-  "vibrator",
+  "vermillion",
   "victim",
   "vignette",
   "villainy",
@@ -344,25 +438,30 @@ static const char *words[] =
   "warehouse",
   "waste",
   "waveform",
+  "Wendover",
   "whiffle ball",
   "whorl",
   "windmill",
+  "Winchester",
   "words",
   "worm",
   "worship",
   "worship",
-  "Xanax",
+  "Wyoming",
   "Xerxes",
   "Xhosa",
   "xylophone",
   "yellow",
+  "Yerington",
   "yesterday",
   "your nose",
   "Zanzibar",
   "zeal",
   "zebra",
   "zest",
-  "zinc"
+  "zettabyte",
+  "zinc",
+  "Zulu"
 };
 
 #define WORD_COUNT (sizeof(words) / sizeof(char *))
